src/refract/JSONSchemaVisitor.cc

```diff
@@ -151,14 +151,14 @@ namespace refract
     template<typename T>
     void JSONSchemaVisitor::primitiveType(const T& e)
     {
-        const typename T::ValueType *v = GetValue<T>(e);
+        const typename T::ValueType *value = GetValue<T>(e);
 
-        if (v) {
+        if (value) {
             setPrimitiveType(e);
 
-            if (fixed && !e.empty()) {
```

Comment:
Instead, we've already checked for `value`. The value might be a sample or default value instead of the `e.content`.